web/frontend/src/components/ResultsDisplay.tsx

```
@@ -586,12 +675,32 @@ const ResultsDisplay: React.FC<ResultsDisplayProps> = ({ results, showDetailedRe
+                      ? 'border-blue-500 bg-blue-50' 
+                      : 'border-gray-300'
+                } shadow-sm text-sm leading-4 font-medium rounded-md ${
+                  bookmarkingIds.has(index)
+                    ? 'text-gray-400 cursor-not-allowed'
+                    : isBookmarked(result) 
+                      ? 'text-blue-700' 
+                      : 'text-gray-700'
+                } bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 disabled:hover:bg-gray-100`}>
+                {bookmarkingIds.has(index) ? (
+                  <svg className="h-4 w-4 mr-1.5 animate-spin" fill="none" viewBox="0 0 24 24">
+                    <circle className="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" strokeWidth="4"></circle>
```

Biome (1.9.4)

[error] Alternative text title element cannot be empty

For accessibility purposes, SVGs should have an alternative text, provided via title element. If the svg element has role="img", you should add the aria-label or aria-labelledby attribute.

(lint/a11y/noSvgWithoutTitle)